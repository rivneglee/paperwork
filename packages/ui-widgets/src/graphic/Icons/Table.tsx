/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
    <path
      d="M888 64H136c-39.6 0-72 32.4-72 72v752c0 39.6 32.4 72 72 72h752c39.6 0 72-32.4 72-72V136c0-39.6-32.4-72-72-72zM340.7 888H136.2l-0.1-0.1V737h204.7v151z m0-223H136V514h204.7v151z m0-223H136V292.1l0.1-0.1h204.5v150z m273.6 446H412.6V737h201.7v151z m0-223H412.7V514h201.7v151z m0-223H412.7V292h201.7v150zM888 887.9l-0.1 0.1H686.4V737H888v150.9z m0-222.9H686.3V514H888v151z m0-223H686.3V292h201.5l0.1 0.1 0.1 149.9z"
    ></path>
  </svg>
);
