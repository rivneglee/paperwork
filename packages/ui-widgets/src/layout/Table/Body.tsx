import React, { FunctionComponent } from 'react';

interface Props {}

export type BodyComponent = FunctionComponent<Props>;

const Body: BodyComponent = ({ children }) => (
  <div className="pw-table__body">
    { children }
  </div>
);

export default Body;
