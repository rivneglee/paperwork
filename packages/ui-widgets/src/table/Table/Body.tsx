import React, { FunctionComponent } from 'react';

import classNames from 'classnames';

interface Props {
  className?: string;
}

export type BodyComponent = FunctionComponent<Props>;

const Body: BodyComponent = ({ children, className }) => (
  <div className={classNames('pw-table__body', className)}>
    { children }
  </div>
);

export default Body;
