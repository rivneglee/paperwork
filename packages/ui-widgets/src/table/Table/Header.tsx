import React, { FunctionComponent } from 'react';

import classNames from 'classnames';

interface Props {
  className?: string;
}

export type HeaderComponent = FunctionComponent<Props>;

const Header: HeaderComponent = ({ children, className }) => (
  <div className={classNames('pw-table__header', className)}>
    <div className="pw-table__row">
      {children}
    </div>
  </div>
);

export default Header;
