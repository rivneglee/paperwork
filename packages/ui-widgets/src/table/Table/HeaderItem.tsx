import React, { FunctionComponent } from 'react';

import classNames from 'classnames';

interface Props {
  className?: string;
  style?: any;
  hideLabel?: boolean;
}

export type HeaderItemComponent = FunctionComponent<Props>;

const HeaderItem: HeaderItemComponent = ({ children, className, hideLabel, style = {} }) => {
  const { width } = style;
  const styleOverrides = width ? { ...style, flex: 'none' } : style;
  return (
    <div className={classNames(
      'pw-table__item',
      className,
    )} style={styleOverrides}>
      {!hideLabel && children}
    </div>
  );
};

export default HeaderItem;
