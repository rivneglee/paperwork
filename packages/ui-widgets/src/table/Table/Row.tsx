import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface Props {
  className?: string;
  onClick?: () => void;
}

export type RowComponent = FunctionComponent<Props>;

const Row: RowComponent = ({ children, className, onClick }) => (
  <div onClick={onClick} className={classNames('pw-table__row', className)}>
    {children}
  </div>
);

export default Row;
