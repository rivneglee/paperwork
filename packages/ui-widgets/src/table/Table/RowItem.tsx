import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface Props {
  columnName?: string;
  className?: string;
  style?: any;
  hideLabel?: boolean;
  textWrap?: string;
}

export type RowItemComponent = FunctionComponent<Props>;

const RowItem: RowItemComponent = ({ children, columnName, className, style = {}, textWrap = 'nowrap', hideLabel }) => {
  const { width } = style;
  const styleOverrides = width ? { ...style, flex: 'none' } : style;
  return (
    <div className={classNames(
      'pw-table__item',
      textWrap && `pw-table_item--${textWrap}`,
      className,
      hideLabel && 'pw-table__item--hide-label',
    )} data-label={columnName} style={styleOverrides}>
      {children}
    </div>
  );
};

export default RowItem;
