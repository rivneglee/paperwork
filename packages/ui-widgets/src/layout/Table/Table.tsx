import React, { FunctionComponent } from 'react';

import Body, { BodyComponent } from './Body';
import Row, { RowComponent } from './Row';
import Header, { HeaderComponent } from './Header';
import RowItem, { RowItemComponent } from './RowItem';
import HeaderItem, { HeaderItemComponent } from './HeaderItem';

interface Props {}

interface TableComponent extends FunctionComponent<Props> {
  Body: BodyComponent;
  Row: RowComponent;
  Header: HeaderComponent;
  RowItem: RowItemComponent;
  HeaderItem: HeaderItemComponent;
}

const Table: TableComponent = ({ children }) => (
  <div className="pw-table__container">
    { children }
  </div>
);

Table.Body = Body;
Table.Row = Row;
Table.Header = Header;
Table.RowItem = RowItem;
Table.HeaderItem = HeaderItem;

export default Table;
