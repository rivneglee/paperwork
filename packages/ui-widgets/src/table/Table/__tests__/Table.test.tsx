import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Table from '../Table';

describe('Table', () => {
  let wrapper: ShallowWrapper;
  beforeAll(() => {
    wrapper = shallow(
      <Table>
        <Table.Header>
          <Table.RowItem>Column A</Table.RowItem>
          <Table.RowItem>Column B</Table.RowItem>
        </Table.Header>
        <Table.Row>
          <Table.RowItem columnName="Column A">A</Table.RowItem>
          <Table.RowItem columnName="Column B">B</Table.RowItem>
        </Table.Row>
      </Table>,
    );
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
