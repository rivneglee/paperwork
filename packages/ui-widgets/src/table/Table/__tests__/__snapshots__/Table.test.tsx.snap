// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render properly 1`] = `
<div
  className="pw-table__container"
>
  <Header>
    <RowItem>
      Column A
    </RowItem>
    <RowItem>
      Column B
    </RowItem>
  </Header>
  <Row>
    <RowItem
      columnName="Column A"
    >
      A
    </RowItem>
    <RowItem
      columnName="Column B"
    >
      B
    </RowItem>
  </Row>
</div>
`;
